src/ProjectFileManager.cpp

```diff
@@ -1204,6 +1207,55 @@ bool ImportProject(AudacityProject &dest, const FilePath &fileName)
 
    return true;
 }
+
+class ImportProgress final
+   : public ImportProgressListener
+{
+public:
+   
+   bool OnImportFileOpened(ImportFileHandle& importFileHandle) override
+   {
+      mImportFileHandle = &importFileHandle;
+      // File has more than one stream - display stream selector
+      if (importFileHandle.GetStreamCount() > 1)
+      {
+         ImportStreamDialog ImportDlg(&importFileHandle, NULL, -1, XO("Select stream(s) to import"));
+
+         if (ImportDlg.ShowModal() == wxID_CANCEL)
+            return false;
+      }
+      // One stream - import it by default
+      else
+         importFileHandle.SetStreamUsage(0,TRUE);
+      return true;
+   }
+   
+   void OnImportProgress(double progress) override
+   {
+      constexpr double ProgressSteps { 1000.0 };
+      if(!mProgressDialog)
+      {
+         wxFileName ff( mImportFileHandle->GetFilename() );
```

Comment:
Should I worry about the safety of that `->`? Should there be a shared or weak pointer?  Or should there be comments in the base class header explaining the assumptions that derived classes may make about lifetimes of `ImportFileHandle`?

Reply:
Aren't comments for `ImportProgressListener` enough?